pkg/tempo/config.go

```diff
@@ -167,10 +169,61 @@ func (c *InstanceConfig) otelConfig() (*configmodels.Config, error) {
 		retryConfig["max_elapsed_time"] = "60s"
 	}
 
-	otelMapStructure["exporters"] = map[string]interface{}{
-		"otlp": otlpExporter,
+	return otlpExporter, nil
+}
+
+// exporters builds one or multiple exporters from a remote_write block.
+// It also supports building an exporter from push_config.
+func (c *InstanceConfig) exporters() (map[string]interface{}, error) {
+	if len(c.RemoteWrite) == 0 {
+		otlpExporter, err := c.exporter(c.PushConfig)
+		return map[string]interface{}{
+			"otlp": otlpExporter,
+		}, err
+	}
+
+	exporters := map[string]interface{}{}
+	for i, remoteWriteConfig := range c.RemoteWrite {
+		exporter, err := c.exporter(remoteWriteConfig)
+		if err != nil {
+			return nil, err
+		}
+		exporterName := fmt.Sprintf("otlp/%d", i)
+		exporters[exporterName] = exporter
+	}
+	return exporters, nil
+}
+
+func (c *InstanceConfig) otelConfig() (*configmodels.Config, error) {
+	otelMapStructure := map[string]interface{}{}
+
+	if len(c.Receivers) == 0 {
+		return nil, errors.New("must have at least one configured receiver")
+	}
+
+	if len(c.RemoteWrite) == 0 && len(c.PushConfig.Endpoint) == 0 {
+		return nil, errors.New("must have a configured at least one remote_write or push_config.endpoint")
+	}
+
+	if len(c.RemoteWrite) != 0 && len(c.PushConfig.Endpoint) != 0 {
+		return nil, errors.New("must not configure push_config and remote_write. push_config is deprecated in favor of remote_write")
 	}
 
+	if c.Batch != nil && c.PushConfig.Batch != nil {
```

Comment:
we need to move batch out of the PushConfig and onto the InstanceConfig. InstanceConfig represents a single otel processor pipeline and we only want one batch processor per pipeline.

with these changes PushConfig represents an otel exporter.

Reply:
Oh totally.  Right now it is staying there to support backwards compatibility with `push_config`.  It is meant to be removed in the future.  If we want to remove it now, I'm all for it.

This check is to make sure only one of the configs is being used.

Reply:
I mean that RemoteWrite is a slice of PushConfigs which represents a set of exporters on a single OTel pipeline.  Batch is only supported at the pipeline level and not at the exporter level. So we need to move Batch out of the PushConfig and onto the InstanceConfig.